File: TUnit.Templates.Tests/Snapshots/InstantiationTestWithVB.TUnit.VB._.verified/TUnit.VB/Tests2.vb
Imports System
Imports TUnit.Core

<Arguments("Hello")>
<Arguments("World")>
Public Class MoreTests
    Private ReadOnly _title As String

    Public Sub New(title As String)
        _title = title
    End Sub

    <Test>
    Public Sub ClassLevelDataRow()
        Console.WriteLine(_title)
        Console.WriteLine("Did I forget that data injection works on classes too?")
    End Sub

    <ClassDataSource(GetType(DataClass), Shared:=SharedType.PerTestSession)>
    Public Property DataClass As DataClass

    <Test>
    <MatrixDataSource>
    Public Sub Matrices(
        <Matrix(1, 2, 3)> a As Integer,
        <Matrix(True, False)> b As Boolean,
        <Matrix("A", "B", "C")> c As String)
        Console.WriteLine("A new test will be created for each data row, whether it's on the class or method level!")
        Console.WriteLine("Oh and this is a matrix test. That means all combinations of inputs are attempted.")
    End Sub

End Class

File: tests/TUnit.Templates.Tests/Snapshots/InstantiationTestWithFSharp.TUnit.FSharp._.verified/TUnit.FSharp/Tests2.fs
namespace TUnit.FSharp

open System
open TUnit.Core

[<Arguments("Hello")>]
[<Arguments("World")>]
type MoreTests(title: string) =

    [<Test>]
    member _.ClassLevelDataRow() =
        Console.WriteLine(title)
        Console.WriteLine("Did I forget that data injection works on classes too?")

    [<ClassDataSource(typeof<DataClass>, Shared = [|SharedType.PerTestSession|])>]
    member val DataClass: DataClass = Unchecked.defaultof<_> with get, set

    [<Test>]
    [<MatrixDataSource>]
    member _.Matrices
        (
            [<Matrix(1, 2, 3)>] a: int,
            [<Matrix(true, false)>] b: bool,
            [<Matrix("A", "B", "C")>] c: string
        ) =
        Console.WriteLine("A new test will be created for each data row, whether it's on the class or method level!")
        Console.WriteLine("Oh and this is a matrix test. That means all combinations of inputs are attempted.")